### PR TITLE
Add new type to add_point_scalar_labels

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

We cannot set numpy.ndarray to use  `add_point_scalar_labels`.

 

### Details

- `numpy.ndarray` does not have property `point_data`.
https://github.com/pyvista/pyvista/blob/66dfce599b78a156692d61e66565b43e4252b608/pyvista/plotting/plotting.py#L4723

### CODE REVIEW

**Review by akaszynski — APPROVED** [review]
LGTM.

**Comment by akaszynski** [comment]
> IMO, this method should instead be changed to support an ndarray of points

Agree with @banesullivan. Let's get this to work with any sequence like the rest of our methods.

**Comment by MatthewFlamm** [comment]
> > IMO, this method should instead be changed to support an ndarray of points

> Agree with @banesullivan. Let's get this to work with any sequence like the rest of our methods.

We have this helper for this occasion: 

https://github.com/pyvista/pyvista/blob/1e9028be8682917833c5b14eccfb0bbaf0da9623/pyvista/utilities/arrays.py#L10-L13

File: pyvista/plotting/plotting.py [hunk 1]
```diff
@@ -4712,15 +4714,24 @@ def add_point_scalar_labels(self, points, labels, fmt=None, preamble='', **kwarg
             VTK label actor.  Can be used to change properties of the labels.
 
         """
-        if not is_pyvista_dataset(points):
-            raise TypeError(f'input points must be a pyvista dataset, not: {type(points)}')
-        if not isinstance(labels, str):
-            raise TypeError('labels must be a string name of the scalars array to use')
+        if not is_pyvista_dataset(points) and not isinstance(points, np.ndarray):
```
Comment (tkoyama010) [inline pyvista/plotting/plotting.py:4717]:
```suggestion
        if not is_pyvista_dataset(points) or not isinstance(points, np.ndarray):
```

File: pyvista/plotting/plotting.py [hunk 2]
```diff
@@ -4712,15 +4714,24 @@ def add_point_scalar_labels(self, points, labels, fmt=None, preamble='', **kwarg
...
             fmt = self._theme.font.fmt
         if fmt is None:
             fmt = '%.6e'
-        scalars = points.point_data[labels]
+        if isinstance(points, np.ndarray):
+            scalars = labels
+        elif is_pyvista_dataset(points):
+            scalars = points.point_data[labels]
+        else:
+            raise TypeError(f'Points type not usable: {type(points)}')
```
Comment (tkoyama010) [inline pyvista/plotting/plotting.py:4734]:
```suggestion
```

File: pyvista/plotting/plotting.py [hunk 3]
```diff
@@ -4712,15 +4714,22 @@ def add_point_scalar_labels(self, points, labels, fmt=None, preamble='', **kwarg
             VTK label actor.  Can be used to change properties of the labels.
 
         """
-        if not is_pyvista_dataset(points):
-            raise TypeError(f'input points must be a pyvista dataset, not: {type(points)}')
-        if not isinstance(labels, str):
-            raise TypeError('labels must be a string name of the scalars array to use')
+        if not is_pyvista_dataset(points) or not isinstance(points, np.ndarray):
```
Comment (tkoyama010) [inline pyvista/plotting/plotting.py:4717]:
```suggestion
        if not is_pyvista_dataset(points) and not isinstance(points, np.ndarray):
```

File: tests/plotting/test_plotting.py [hunk 4]
```diff
@@ -3015,3 +3015,18 @@ def test_add_point_scalar_labels_fmt():
...
+
+
+def test_add_point_scalar_labels_list():
+    plotter = pyvista.Plotter()
+
+    points = np.array([[0, 0, 0], [1, 0, 0], [0, 1, 0], [1, 1, 0], [0.5, 0.5, 0.5], [1, 1, 1]])
+    labels = [0.0, 1.0, 2.0, 3.0, 4.0, 5.0]
+
+    with pytest.raises(TypeError):
+        plotter.add_point_scalar_labels(points, points=False)
```
Comment (tkoyama010) [inline tests/plotting/test_plotting.py:3027]:
```suggestion
        plotter.add_point_scalar_labels(points=False)
```

**Comment by tkoyama010** [comment]
@MatthewFlamm Thanks. It's a great helper.

**Review by banesullivan — APPROVED** [review]

**Pull request merged**